app/src/main/java/appricottsoftware/clarity/RecyclerAdapter.java

```diff
@@ -0,0 +1,151 @@
+        this.isChannelView = isChannelView;
+    }
+
+    @Override
+    public CustomViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        View theView = LayoutInflater.from(parent.getContext())
+                .inflate(R.layout.single_recycle_view_item, parent, false);
+        return new CustomViewHolder(theView);
+
+    }
+
+    @Override
+    public void onBindViewHolder(CustomViewHolder holder, int position) {
+        RecyclerListItem recyclerListItem = recyclerList_ItemList.get(position);
+
+        holder.textViewTitle.setText(recyclerListItem.getTitle());
+        Glide.with(theContext).load(recyclerListItem.getImage()).into(holder.ImageViewAlbum);
+
+        holder.itemView.setBackgroundColor(selected_position == position ? Color.LTGRAY : Color.TRANSPARENT);
+    }
+
+    @Override
+    public int getItemCount() {
+        return recyclerList_ItemList.size();
+    }
+
+    public class CustomViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
+
+        public ImageView ImageViewAlbum;
+        public TextView textViewTitle;
+
+        public CustomViewHolder(View itemView) {
+            super(itemView);
+            itemView.setOnClickListener(this);
+
+            textViewTitle = itemView.findViewById(R.id.textView_title);
+            ImageViewAlbum = itemView.findViewById(R.id.imageView_album);
+
+        }
+
+        @Override
+        public void onClick(final View view) {
+            notifyItemChanged(selected_position);
+            selected_position = getLayoutPosition();
+            notifyItemChanged(selected_position);
+
+            if (isChannelView){
+
+                if(view.getContext() instanceof PlayerInterface) {
+                    playerInterface = (PlayerInterface) view.getContext();
+                } else {
+                    Log.e(TAG, view.getContext().toString() + " must implement PlayerInterface");
+                    throw new ClassCastException(view.getContext().toString() + " must implement PlayerInterface");
+                }
+
+                //Channel epToPlay = recyclerList_ItemList.get(selected_position).getClass();
+
+                Episode testEpisode = Episode.getSampleEpisode();
+                playerInterface.playEpisode(testEpisode);
+            }
+            else{
+                Toast.makeText(view.getContext(),"CHANNEL ADDED TO DATABASE AND CHANNEL LIST", Toast.LENGTH_SHORT).show();
+
+                int uid = 1;
+                String title = recyclerList_ItemList.get(selected_position).getTitle();
+                String imageURL = recyclerList_ItemList.get(selected_position).getImage();
+
+                ClarityApp.getRestClient().createChannel(uid, title, imageURL, view.getContext(), new JsonHttpResponseHandler() {
+                    @Override
+                    public void onSuccess(int statusCode, Header[] headers, JSONObject response) {
+                        Log.e(TAG, "onSuccess1 : " + response.toString() );
```

Comment:
Is this onSuccess response an error? If not, consider using Log.v.

Reply:
Removed.